### DISCUSSION TITLE
I can't save a buffer, I keep getting permission denied

### DISCUSSION THREAD

**Discussion by CaptainEboy** [termux/termux-packages#8417] [Q&A] [answered]
Can't save a buffer, keep getting permission denied like in the image below 👇👇

![Screenshot_20220101-034009_1641009464435](https://user-images.githubusercontent.com/63905637/147843664-69ef587f-b580-4896-a99b-d39aef4283fc.jpg)

**Comment by CaptainEboy** [comment]
Any solution to the above problem, would be greatly appreciated.

Thanks

**Reply by sk0kanik** [reply]
Note:
* External sdcard is readonly on Termux.
* Internal storage is writable but you need to give [storage permission](https://wiki.termux.com/wiki/Termux-setup-storage).

**Reply by CaptainEboy** [reply]
Alright, I actually didn't use the internal storage cause I didn't want it to mess with my device storage.

Would check it out.

Thanks 💖🔥

**Comment by CaptainEboy** [comment, marked as answer]
Alright.

Thanks @sk0kanik 

The solution to the problem was actually to access the internal sdcard & not the external sdcard(reason provided by @sk0kanik )

Accessing the internal sdcard instead of the external sdcard solves the problem.

However, I am thinking of turning this post/question to a pull request so that the termux team can provide the write-access feature for external sdcard in a future update
